cmd/kubeadm/app/cmd/phases/addons.go

```diff
@@ -17,186 +17,108 @@ limitations under the License.
-			cmd.Flags().StringVar(&cfg.Networking.PodSubnet, "pod-network-cidr", cfg.Networking.PodSubnet, `The range of IP addresses used for the Pod network`)
-		}
-
-		if properties.use == "all" || properties.use == "coredns" {
-			cmd.Flags().StringVar(&cfg.Networking.DNSDomain, "service-dns-domain", cfg.Networking.DNSDomain, `Alternative domain for services`)
-			cmd.Flags().StringVar(&cfg.Networking.ServiceSubnet, "service-cidr", cfg.Networking.ServiceSubnet, `The range of IP address used for service VIPs`)
-			cmd.Flags().StringVar(&featureGatesString, "feature-gates", featureGatesString, "A set of key=value pairs that describe feature gates for various features. "+
-				"Options are:\n"+strings.Join(features.KnownFeatures(&features.InitFeatureGates), "\n"))
-		}
-		subCmds = append(subCmds, cmd)
+// runEnsureKubeProxyAddon installs KubeProxy addon to a Kubernetes cluster
+func runEnsureKubeProxyAddon(c workflow.RunData) error {
+	cfg, client, err := getAddonData(c)
+	if err != nil {
+		return err
 	}
-
-	return subCmds
+	return proxyaddon.EnsureProxyAddon(cfg, client)
 }
 
-// runAddonsCmdFunc creates a cobra.Command Run function, by composing the call to the given cmdFunc with necessary additional steps (e.g preparation of input parameters)
-func runAddonsCmdFunc(cmdFunc func(cfg *kubeadmapi.InitConfiguration, client clientset.Interface) error, cfg *kubeadmapiv1beta1.InitConfiguration, kubeConfigFile string, cfgPath *string, featureGatesString *string) func(cmd *cobra.Command, args []string) {
-
-	// the following statement build a clousure that wraps a call to a cmdFunc, binding
-	// the function itself with the specific parameters of each sub command.
-	// Please note that specific parameter should be passed as value, while other parameters - passed as reference -
-	// are shared between sub commands and gets access to current value e.g. flags value.
-
-	return func(cmd *cobra.Command, args []string) {
-		var err error
-		if err := validation.ValidateMixedArguments(cmd.Flags()); err != nil {
-			kubeadmutil.CheckErr(err)
-		}
-
-		if cfg.FeatureGates, err = features.NewFeatureGate(&features.InitFeatureGates, *featureGatesString); err != nil {
-			kubeadmutil.CheckErr(err)
-		}
-
-		internalcfg := &kubeadmapi.InitConfiguration{}
-		kubeadmscheme.Scheme.Convert(cfg, internalcfg, nil)
-		kubeConfigFile = cmdutil.FindExistingKubeConfig(kubeConfigFile)
-		client, err := kubeconfigutil.ClientSetFromFile(kubeConfigFile)
-		kubeadmutil.CheckErr(err)
-		internalcfg, err = configutil.ConfigFileAndDefaultsToInternalConfig(*cfgPath, cfg)
-		kubeadmutil.CheckErr(err)
-		if err := features.ValidateVersion(features.InitFeatureGates, internalcfg.FeatureGates, internalcfg.KubernetesVersion); err != nil {
-			kubeadmutil.CheckErr(err)
-		}
-
-		// Execute the cmdFunc
-		err = cmdFunc(internalcfg, client)
-		kubeadmutil.CheckErr(err)
+func getAddonPhaseFlags(name string) []string {
+	flags := []string{
```

Comment:
it seems that some flag are missing:
- kubernetes-version
- image respository (that if I'm not wrong should be added to init [here](https://github.com/kubernetes/kubernetes/blob/b8fece50f533bdfbf58b638f73e18a5773c3434c/cmd/kubeadm/app/cmd/init.go#L168-L172) by creating `options.AddImageMetaFlags`)